R/LearnerSurvXGBoost.R

```diff
@@ -0,0 +1,158 @@
+        ParamInt$new("early_stopping_rounds", default = NULL, lower = 1L, special_vals = list(NULL), tags = "train"),
+        ParamLgl$new("maximize", default = NULL, special_vals = list(NULL), tags = "train"),
+        ParamFct$new("sample_type", default = "uniform", levels = c("uniform", "weighted"), tags = "train"),
+        ParamFct$new("normalize_type", default = "tree", levels = c("tree", "forest"), tags = "train"),
+        ParamDbl$new("rate_drop", default = 0, lower = 0, upper = 1, tags = "train"),
+        ParamDbl$new("skip_drop", default = 0, lower = 0, upper = 1, tags = "train"),
+        ParamLgl$new("one_drop", default = FALSE, tags = "train"),
+        ParamFct$new("tree_method", default = "auto", levels = c("auto", "exact", "approx", "hist", "gpu_hist"), tags = "train"),
+        ParamFct$new("grow_policy", default = "depthwise", levels = c("depthwise", "lossguide"), tags = "train"),
+        ParamInt$new("max_leaves", default = 0L, lower = 0L, tags = "train"),
+        ParamInt$new("max_bin", default = 256L, lower = 2L, tags = "train"),
+        ParamUty$new("callbacks", default = list(), tags = "train"),
+        ParamDbl$new("sketch_eps", default = 0.03, lower = 0, upper = 1, tags = "train"),
+        ParamDbl$new("scale_pos_weight", default = 1, tags = "train"),
+        ParamUty$new("updater", tags = "train"), # Default depends on the selected booster
+        ParamLgl$new("refresh_leaf", default = TRUE, tags = "train"),
+        ParamFct$new("feature_selector", default = "cyclic", levels = c("cyclic", "shuffle", "random", "greedy", "thrifty"), tags = "train"),
+        ParamInt$new("top_k", default = 0, lower = 0, tags = "train"),
+        ParamFct$new("predictor", default = "cpu_predictor", levels = c("cpu_predictor", "gpu_predictor"), tags = "train")
+      ))
+      # param deps
+      ps$add_dep("print_every_n", "verbose", CondEqual$new(1L))
+      ps$add_dep("sample_type", "booster", CondEqual$new("dart"))
+      ps$add_dep("normalize_type", "booster", CondEqual$new("dart"))
+      ps$add_dep("rate_drop", "booster", CondEqual$new("dart"))
+      ps$add_dep("skip_drop", "booster", CondEqual$new("dart"))
+      ps$add_dep("one_drop", "booster", CondEqual$new("dart"))
+      ps$add_dep("tree_method", "booster", CondAnyOf$new(c("gbtree", "dart")))
+      ps$add_dep("grow_policy", "tree_method", CondEqual$new("hist"))
+      ps$add_dep("max_leaves", "grow_policy", CondEqual$new("lossguide"))
+      ps$add_dep("max_bin", "tree_method", CondEqual$new("hist"))
+      ps$add_dep("sketch_eps", "tree_method", CondEqual$new("approx"))
+      ps$add_dep("feature_selector", "booster", CondEqual$new("gblinear"))
+      ps$add_dep("top_k", "booster", CondEqual$new("gblinear"))
+      ps$add_dep("top_k", "feature_selector", CondAnyOf$new(c("greedy", "thrifty")))
+
+      ps$values = list(nrounds = 1L, verbose = 0L, objective = "survival:cox",
+        eval_metric = "cox-nloglik")
```

Comment:
If this is the way that mlr3 team usually do this then okay, but the problem with changing the defaults like this is that if the user calls `lrn(surv.xgboost, verbose = 1)/lrn$vals = list(verbose = 1)` for example, then the objective reverts to default. Which is problematic as (correct me if wrong) it *must* be `survival:cox` for it to work here. So it may be better to change this within `.train` so the user cannot overwrite it by mistake.

Reply:
good point, objective is now in `.train`. For `eval_metric` I'm not sure. People might want to optimize w.r.t. to something different then cox loglik